From 96a5be8ce70cfcc4697f370a90ba03e786b4facb Mon Sep 17 00:00:00 2001
From: Mike Zeller <mike.zeller@joyent.com>
Date: Tue, 3 Apr 2018 21:55:37 +0000
Subject: [PATCH] TRITON-301 TRITON-58 created a circular import dep between
 lib/cloudapi2.js and lib/common.js Reviewed by: Trent Mick <trentm@gmail.com>
 Reviewed by: Dave Eddy <dave.eddy@joyent.com> Approved by: Trent Mick
 <trentm@gmail.com>

---
 lib/cloudapi2.js             | 7 -------
 lib/common.js                | 2 +-
 lib/constants.js             | 9 ++++++++-
 lib/do_instance/do_create.js | 2 +-
 4 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/lib/cloudapi2.js b/lib/cloudapi2.js
index 98fcacb..94c57eb 100644
--- a/lib/cloudapi2.js
+++ b/lib/cloudapi2.js
@@ -154,13 +154,6 @@ function CloudApi(options) {
     this.client = new SaferJsonClient(options);
 }
 
-// <Network Object Key> -> <expected typeof>
-CloudApi.prototype.NETWORK_OBJECT_FIELDS = {
-    ipv4_uuid: 'string',
-    ipv4_ips: 'string'
-};
-
-
 CloudApi.prototype.close = function close(callback) {
     this.log.trace({host: this.client.url && this.client.url.host},
         'close cloudapi http client');
diff --git a/lib/common.js b/lib/common.js
index 04aee41..613732e 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -25,7 +25,7 @@ var wordwrap = require('wordwrap');
 var errors = require('./errors'),
     InternalError = errors.InternalError;
 var NETWORK_OBJECT_FIELDS =
-    require('./cloudapi2').CloudApi.prototype.NETWORK_OBJECT_FIELDS;
+    require('./constants').NETWORK_OBJECT_FIELDS;
 
 
 // ---- support stuff
diff --git a/lib/constants.js b/lib/constants.js
index 5389b0e..18fe901 100644
--- a/lib/constants.js
+++ b/lib/constants.js
@@ -46,11 +46,18 @@ if (process.env.TRITONTEST_CLI_CONFIG_DIR) {
     CLI_CONFIG_DIR = mod_path.resolve(process.env.HOME, '.triton');
 }
 
+// <Network Object Key> -> <expected typeof>
+var NETWORK_OBJECT_FIELDS = {
+    ipv4_uuid: 'string',
+    ipv4_ips: 'string'
+};
+
 
 // ---- exports
 
 module.exports = {
-    CLI_CONFIG_DIR: CLI_CONFIG_DIR
+    CLI_CONFIG_DIR: CLI_CONFIG_DIR,
+    NETWORK_OBJECT_FIELDS: NETWORK_OBJECT_FIELDS
 };
 
 
diff --git a/lib/do_instance/do_create.js b/lib/do_instance/do_create.js
index 12463a0..eb5ef38 100644
--- a/lib/do_instance/do_create.js
+++ b/lib/do_instance/do_create.js
@@ -20,7 +20,7 @@ var distractions = require('../distractions');
 var errors = require('../errors');
 var mat = require('../metadataandtags');
 var NETWORK_OBJECT_FIELDS =
-    require('../cloudapi2').CloudApi.prototype.NETWORK_OBJECT_FIELDS;
+    require('../constants').NETWORK_OBJECT_FIELDS;
 
 function parseVolMount(volume) {
     var components;
-- 
2.21.0

